apps/dashboard/app/controllers/support_ticket_controller.rb

```diff
@@ -0,0 +1,45 @@
+class SupportTicketController < ApplicationController
+
+  def new
+    support_service = create_service_class
+    @support_ticket = support_service.default_support_ticket(params)
+
+    render get_ui_template
+  end
+
+  def create
+    read_support_ticket_from_request
+
+    support_service = create_service_class
+    @support_ticket = support_service.validate_support_ticket(read_support_ticket_from_request)
+
+    if @support_ticket.invalid? || !@support_ticket.errors.empty?
+      flash.now[:alert] = t('dashboard.support_ticket.validation_error')
+      render get_ui_template
+      return
+    end
+
+    support_ticket_response = support_service.deliver_support_ticket(@support_ticket)
+    redirect_to root_url, :flash => { :notice => support_ticket_response }
+
+    rescue => error
+      logger.error "Could not create support ticket. support_service=#{support_service} error=#{error}"
+      flash.now[:alert] = t('dashboard.support_ticket.generic_error', error: error)
+      render get_ui_template
+  end
+
+  private
+
+  def create_service_class
+    class_name = ::Configuration.support_ticket_config.fetch(:service, "SupportTicketEmailService")
+    class_name.constantize.new
+  end
+
+  def read_support_ticket_from_request
+    params.require(:support_ticket).permit!
+  end
+
+  def get_ui_template
+    ::Configuration.support_ticket_config.fetch(:ui_template, "new")
+  end
```

Comment:
I know we talked about additional templates but I'm wondering if they'll ever be used and if we need to add them now. 

You're going to want something for `rt` ticketing system and it can have it's own template - separate from the email template - but I am concerned the UI template is never going to be used.

The actual email template, I could se being useful for some sites, but I don't really expect anyone to add to the form template.

In the case of emails, what additional fields _could_ they add?

Reply:
> In the case of emails, what additional fields could they add?

Additional form fields could either be consolidated into the email message body, or inserted into the message headers.

Among some of the additional fields we collect in our ticketing system are: Priority, Severity, Affiliation, School, Department, and Problem Type. We have set up at least 3 different web forms for support ticket intake for our different sub-groups, and each of them specify different fields relevant to their service area.

Reply:
OK - it may be an undocumented feature. I sorta shudder at the thought of communicating this to someone who likely doesn't know html to build the form, ruby to understand the variables and Rails to put it all together (with a new text template too!).

I'll guess the documented best practice will be to prompt all that stuff in the default description so folks can just fill it out in the text area. 

As the support staff, something like this is almost untenable just because of the levels & spans of knowledge required to carry it out.

Reply:
Thinking about this a bit more, maybe we should defer it and use the form building apparatus we already have.  So they could add new fields through yaml config like so.  

```yaml
support_ticket:
  extra_fields:
    priority:
      widget: select
      options:
        - low
        - medium
        - high
```

Folks are already familiar with this strategy through batch connect forms so it's familiar. We could dump the hash into the template so they may not even need to override the text template either.

Reply:
Will take a look at the `BatchConnect` form building

Reply:
Thinking about this some more, I'm fairly sure you can already override any view template if you drop a replacement here `/etc/ood/config/apps/dashboard/views/`

Reply:
Yes, templates can be overriden using `OOD_LOAD_EXTERNAL_CONFIG=true` (for dev and test, in production is enabled by default) and` /etc/ood/config/apps/dashboard/views/`

We can use this if needed, removing the template override from Controller.

Reply:
@johrstrom I had a look at the `HTML5` frontend validation and the `BatchConnect` dynamic form building used for interactive apps.

**Validation:**
The required field and email validation for HTML5 is quite trivial to implement. As for the attachments, to simplify the javascript, we could add a single attachments field that can select multiple files. This will remove the need for the JS code that dynamically adds and removes attachments.

**`BatchConnect` dynamic form building:**
I can update the current implementation to drive the full support ticket form using the dynamic form building. This could be customizable using the configuration object similar to how it is done in `BatchConnect` apps.

Or I could use the `BatchConnect` dynamic form building only for the additional fields if provided.

What do you think?

Having said that, we will use a custom template for the RT API integration.
For this reason, we would like to keep the UI template override as an undocumented feature. This simplifies how the UI template is configured and keeps the support ticket settings in one place.

Reply:
> As for the attachments, ...

I'm fine to have as much javascript as we need for attachments. I just don't want _more than we need_ is all. A check on upload file size isn't unreasonable if you want it here - which is to say I'm fine with a little javascript for stuff that html5 validation can't handle. 

> BatchConnect dynamic form building:

I'm wondering how much of batch connect we actually need. I'm thinking we just read the YAML and make `smart_attributes` to build a form.  I doubt there's any reuse out of batch_connect stuff directly, I'm guessing it's more reuse of smart attributes alone.

> Having said that, we will use a custom template for the RT API integration.

I'm not familiar with RT, but I'd guess someone else could benefit from it. I guess I overlooked/misheard you all in the meeting. I'm happy to have RT support built into OOD if you want to contribute it back. Obviously I can't tell you to do it, but I guess I am asking if you could 😆 .

Reply:
I have made the FE validation + configuration and service class selection changes.

I couldn't move forward with the smart_attributes as I couldn't figure out how to integrate the session data into them.
session_id is a request parameter, passed through in the link from the interactive session card. Then a session object is loaded into the model and used in the template. I couldn't replicate de behaviour with smart attributes and decided to leave it for a second iteration of the feature.

Reply:
This is a demo or the new FE validation:

![support_ticket](https://user-images.githubusercontent.com/17336210/191822657-96624d01-c26b-42c6-8a81-bc7b483bc2d2.gif)

Reply:
This is the new format of the configuration:

```yaml
support_ticket:
  email:
    from: "config@example.com"
    to: "support@support.system.com"
    delivery_method: "smtp"
    raise_delivery_errors: true
    perform_deliveries: true
    delivery_settings:
      address: "host.docker.internal"
      port: "25"
  #ui_template: "rt_service_template"
```